eZ/Publish/Core/FieldType/Image/IO/Legacy.php

```diff
@@ -149,7 +150,13 @@ public function loadBinaryFile($binaryFileId)
     public function loadBinaryFileByUri($binaryFileUri)
     {
         try {
-            return $this->publishedIOService->loadBinaryFileByUri($binaryFileUri);
+            $image = $this->publishedIOService->loadBinaryFileByUri($binaryFileUri);
+
+            if ($image instanceof MissingBinaryFile) {
+                $image = $this->draftIOService->loadBinaryFileByUri($binaryFileUri);
+            }
+
+            return $image;
         } catch (InvalidArgumentException $prefixException) {
             // InvalidArgumentException means that the prefix didn't match, NotFound can pass through
             try {
```

Comment:
AFAIK we don't throw InvalidArgumentException so I will create followup PR for 7.0